File: ISA-Backend/src/main/java/com/example/ISABackend/service/MedicineService.java
package com.example.ISABackend.service;

import com.example.ISABackend.dto.SearchMedicine;
import com.example.ISABackend.model.Medicine;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

public interface MedicineService {

    public Medicine getById(Long id);
    public List<Medicine> getAll();
    public ArrayList<Medicine> search(SearchMedicine searchParameters);
    public ArrayList<Medicine> sort(ArrayList<Medicine> sortMedicines, String sortType);
    public Medicine download(Long medicine_id);
    public Medicine addNew(Medicine newMedicine);
    public Medicine getByCode(int code);
    public Medicine getByName(String name);
    public ArrayList<Long> addReplacements(ArrayList<Long> replacementsId, Long medicineId);

}

File: ISA-Frontend/src/app/ISA/shared/service/medicine.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { User } from "../model/User";
import { Medicine } from "../model/Medicine";
import { SearchMedicine } from "../model/SearchMedicine";


@Injectable()
export class MedicineService {
    constructor(private http: HttpClient) {
    }

    public getAllMedicines():Observable<Medicine[]>{
      return this.http.get<Medicine[]>("/api/medicine");
    }

    public searchMedicine(sm:SearchMedicine):Observable<Medicine[]>{
      return this.http.post<Medicine[]>("/api/medicine/search", sm);
    }

    public sortMedicines(medicineList:Medicine[], sortType:string):Observable<Medicine[]>{
      return this.http.post<Medicine[]>("/api/medicine/sort/"+sortType, medicineList);
    }

    public downloadSpecifications( medicine_id:number){
      return this.http.post<number>("/api/medicine/download/" + medicine_id, medicine_id);
    }
}